lib/actions/os.coffee

```diff
@@ -134,50 +144,125 @@ exports.download =
 			console.info('The image was downloaded successfully')
 		.nodeify(done)
 
+buildConfig = (image, deviceType, advanced = false) ->
+	form = require('resin-cli-form')
+	helpers = require('../utils/helpers')
+
+	helpers.getManifest(image, deviceType)
+	.get('options')
+	.then (questions) ->
+		if not advanced
+			advancedGroup = _.findWhere questions,
+				name: 'advanced'
+				isGroup: true
+
+			if advancedGroup?
+				override = helpers.getGroupDefaults(advancedGroup)
+
+		return form.run(questions, { override })
+
+exports.buildConfig =
+	signature: 'os build-config <image> <device-type>'
+	description: 'build the OS config and save it to the JSON file'
+	help: '''
+		Use this command to prebuild the OS config once and skip the interactive part of `resin os configure`.
+
+		Examples:
+
+			$ resin os build-config ../path/rpi3.img raspberrypi3 --output rpi3-config.json
+			$ resin os configure ../path/rpi3.img 7cf02a6 --config "$(cat rpi3-config.json)"
+	'''
+	permission: 'user'
+	options: [
+		{
+			signature: 'advanced'
+			description: 'show advanced commands'
```

Comment:
OK I'm innocent, I've just copied it from a different place below :P Anyway fixing now